shopify/usage-subscription-template/api/actions/chargeShop.ts

```diff
@@ -1,32 +1,52 @@
-import { ActionOptions, ChargeShopGlobalActionContext } from "gadget-server";
+import { ActionOptions } from "gadget-server";
 import { getCappedAmount } from "../../utilities";
-
-/**
- * @param { ChargeShopGlobalActionContext } context
- */
-export async function run({ params, logger, api, connections }) {
-  const { shop, order } = params;
+import { SubscriptionLineItems } from "../models/shopifyShop/actions/subscribe";
+
+export const run: ActionRun = async ({ params, logger, api, connections }) => {
+  const {
+    shop,
+    order,
+  }: {
+    shop?: {
+      id?: string;
+      currency?: string;
+      overage?: number;
+      activeSubscriptionId?: string;
+      usagePlanId?: string;
+      plan?: {
+        price?: number;
+      };
+    };
+    order?: {
+      email?: string;
+    };
+  } = params;
 
   // Creating an instance of the Shopify Admin API
-  const shopify = await connections.shopify.forShopId(shop.id);
+  const shopify = await connections.shopify.forShopId(shop?.id as string);
```

Comment:
We use type casting throughout this file. Makes me think that we should just check to make sure required values exist before processing, or else throw an error

`if(shop?.id) { /** code */ }` makes all these type issues go away